react-components/src/metadata/MetadataTemplateView.js

```diff
@@ -0,0 +1,543 @@
+                                    parentId={ids.METADATA_TEMPLATE_VIEW}
+                                    onConfirm={() => {
+                                        this.closeConfirmationDialog();
+                                        handleSubmit();
+                                    }}
+                                    onClose={this.closeMetadataTemplateDialog}
+                                    onCancel={this.closeConfirmationDialog}
+                                    title={getMessage("save")}
+                                    dialogContent={getMessage("confirmCloseUnsavedChanges")}
+                                    confirmLabel={getMessage("yes")}
+                                    closeLabel={getMessage("no")}
+                                    cancelLabel={getMessage("cancel")}
+                />
+            </Dialog>
+        );
+    }
+}
+
+const mapPropsToValues = props => {
+    const { template } = props;
+    const metadata = { ...props.metadata };
+
+    const attributeMap = {};
+    const mapAttributesToAttrMap = (attrMap, attributes) => attributes.forEach(attribute => {
+        const attrCopy = { ...attribute };
+        attrMap[attribute.name] = attrCopy;
+
+        if (attrCopy.attributes && attrCopy.attributes.length > 0) {
+            const subAttrMap = {};
+            mapAttributesToAttrMap(subAttrMap, attrCopy.attributes);
+            attrCopy.attributes = subAttrMap;
+        }
+    });
+
+    mapAttributesToAttrMap(attributeMap, template.attributes);
+
+    const mapAttributesToAVUs = (attributes, propsAVUs) => {
+
+        let avus = propsAVUs ? [...propsAVUs] : [];
+
+        attributes.filter((attribute) => attribute.required).forEach((attribute) => {
+            if (avus.filter((avu) => avu.attr === attribute.name).length < 1) {
+                avus.push(newAVU(attribute));
+            }
+
+            const { attributes } = attribute;
+            if (attributes && attributes.length > 0) {
+                avus = avus.map(propsAVU => {
+                    let avu = { ...propsAVU };
+
+                    if (avu.attr === attribute.name) {
+                        avu.avus = mapAttributesToAVUs(attributes, avu.avus);
+                    }
+
+                    return avu;
+                });
+            }
+        });
+
+        return avus;
+    };
+
+    metadata.avus = mapAttributesToAVUs(template.attributes, metadata.avus);
+
+    return { template, attributeMap, metadata };
+};
+
+const validateAVUs = (avus, attributeMap) => {
+    const avuArrayErrors = [];
+
+    avus.forEach((avu, avuIndex) => {
+        const avuErrors = {};
+        const value = avu.value;
+
+        let attrTemplate = attributeMap[avu.attr];
+        if (!attrTemplate) {
+            return;
+        }
+
+        if (attrTemplate.required && value === "") {
+            avuErrors.value = getMessage("required");
+            avuErrors.error = true;
```

Comment:
I ran into an odd feature(?) of Formik (or a bug in my code) that caused a nested child's error to always populate with an empty object `{}`, even when my validation function tried to explicitly set that field's `avuErrors` to `null`. So a simple `error && getMessage("error")` wouldn't work for displaying attribute errors. That's why this validation function sets an explicit `avuErrors.error` flag for the field to examine in its render method.